src/main/java/com/rocketden/main/exception/api/ApiError.java

```diff
@@ -1,22 +1,13 @@
 package com.rocketden.main.exception.api;
 
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
 import org.springframework.http.HttpStatus;
 
-// Error object holding response message and status details
-@Getter
-@EqualsAndHashCode
-public class ApiError {
-    private final HttpStatus status;
-    private final ApiErrorResponse response;
+// Interface representing an API error that holds status and response object
+public interface ApiError {
 
-    public ApiError(HttpStatus status, String message) {
-        this.status = status;
-        this.response = new ApiErrorResponse(message);
-    }
+    HttpStatus getStatus();
 
-    public String getMessage() {
-        return this.response.getMessage();
-    }
+    ApiErrorResponse getResponse();
```

Comment:
I don't think this method is ever used? Thoughts on implementing it in `RoomError` or `UserError`, or else removing it?

Reply:
If you do this, could you test it somewhere as well?

Reply:
Yeah it's kinda unclear since RoomError and UserError don't actually seem to "implement" it, but rather it's auto-generated by the `@Getter` notation (it won't compile if `@Getter` is removed as expected if you inherit an interface without implementing all the methods). This is where the getResponse method is used:

<img width="1031" alt="Where getResponse is used in codebase" src="https://user-images.githubusercontent.com/24768574/93647540-80027000-f9bd-11ea-9a1e-077a0d8d0f03.png">

Reply:
Ok that makes sense, thanks for the explanation. 👍